core/src/util.jl

```diff
@@ -358,35 +358,40 @@ function is_main_network(subnetwork_id::Int32)::Bool
     return subnetwork_id == 1
 end
 
-function get_all_priorities(db::DB, config::Config)::Vector{Int32}
-    priorities = Set{Int32}()
+function get_all_demand_priorities(db::DB, config::Config;)::Vector{Int32}
+    demand_priorities = Set{Int32}()
     is_valid = true
-    # TODO: Is there a way to automatically grab all tables with a priority column?
-    for (type, name) in [
-        (UserDemandStaticV1, "UserDemand / static"),
-        (UserDemandTimeV1, "UserDemand / time"),
-        (LevelDemandStaticV1, "LevelDemand / static"),
-        (LevelDemandTimeV1, "LevelDemand / time"),
-        (FlowDemandStaticV1, "FlowDemand / static"),
-        (FlowDemandTimeV1, "FlowDemand / time"),
-    ]
-        priority_col = load_structvector(db, config, type).priority
-        priority_col = Int32.(coalesce.(priority_col, Int32(0)))
-        if valid_priorities(priority_col, config.allocation.use_allocation)
-            union!(priorities, priority_col)
+
+    for name in names(Ribasim; all = true)
+        type = getfield(Ribasim, name)
+        if !(
+            (type isa DataType) &&
+            type <: Legolas.AbstractRecord &&
+            hasfield(type, :demand_priority)
```

Comment:
While I appreciate this refactor, I'd say the previous code was more readable. I now have to look up all tables with `demand_priority` to get the same understanding. Just a general remark, it's a fine line between preventing duplication bugs and readability.